### PR TITLE
Add additional fields to projects data source

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/5868

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
resource-manager: added additional fields to `google_projects` datasource
```

### CODE REVIEW

**Comment by upodroid** [comment]
One of the fields is failing the acceptance test.

``` 
REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google   master  18⬇  2✎  USAGE  $    make testacc TEST=./google TESTARGS='-run=TestAccDataSourceGoogleProjects_basic'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccDataSourceGoogleProjects_basic -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceGoogleProjects_basic
=== PAUSE TestAccDataSourceGoogleProjects_basic
=== CONT  TestAccDataSourceGoogleProjects_basic
    TestAccDataSourceGoogleProjects_basic: testing.go:674: Step 0 error: Check failed: Check 5/7 error: data.google_projects.my-project: Attribute 'projects.0.parent' expected to be set
--- FAIL: TestAccDataSourceGoogleProjects_basic (3.85s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google/google   7.372s
FAIL
make: *** [testacc] Error 1
```

It is the same data structure as labels which is passing the test.

```
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google   master  20⬇  2✎  USAGE  $    gcloud projects describe ops-mahamed-ali --format json
{
  "createTime": "2019-04-04T15:49:29.369Z",
  "labels": {
    "enviroment": "dev"
  },
  "lifecycleState": "ACTIVE",
  "name": "Some Name",
  "parent": {
    "id": "452031532549",
    "type": "folder"
  },
  "projectId": "REDACTED",
  "projectNumber": "550924169191"
}
```

**Comment by upodroid** [comment]
fixed release note error

File: third_party/terraform/data_sources/data_source_google_projects.go [hunk 1]
```diff
@@ -23,6 +23,37 @@ func dataSourceGoogleProjects() *schema.Resource {
...
 						},
+						"create_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"labels": {
+							Type:        schema.TypeMap,
+							Computed:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Description: `An optional reference to a parent Resource.`,
```
Comment (ndmckinley) [inline third_party/terraform/data_sources/data_source_google_projects.go:34]: I think these descriptions got flipped.

File: third_party/terraform/data_sources/data_source_google_projects.go [hunk 2]
```diff
@@ -79,11 +110,38 @@ func flattenDatasourceGoogleProjectsList(v interface{}) []map[string]interface{}
...
+			mLabels = pLabels
+		}
+		if pLifecycleState, ok := p["lifecycleState"]; ok {
+			mLifecycleState = pLifecycleState
+		}
+		if pCreateTime, ok := p["createTime"]; ok {
+			mCreateTime = pCreateTime
+		}
+		if pParent, ok := p["parent"]; ok {
+			mParent = pParent
```
Comment (ndmckinley) [inline third_party/terraform/data_sources/data_source_google_projects.go:134]: I think something is probably a little mixed up about this typing - if you want to investigate, I'd log the value you're receiving, including type, at p["parent"].

File: third_party/terraform/data_sources/data_source_google_projects.go (hunk 1 above)
Comment (upodroid) [reply third_party/terraform/data_sources/data_source_google_projects.go]: fixed

File: third_party/terraform/data_sources/data_source_google_projects.go (hunk 2 above)
Comment (upodroid) [reply third_party/terraform/data_sources/data_source_google_projects.go:134]: I figured it out, was testing the wrong fields. parent instead parent.id and parent.type.

**Pull request merged**